File: swift/lib/Sources/Message/Param.swift
public class Param: Codable {
    public var n: String
    public var v: AnyCodable?

    private enum CodingKeys: String, CodingKey {
        case n
        case v
    }

    public init(_ n: String, _ v: AnyCodable?) {
        self.n = n
        self.v = v
    }

    public init(_ n: String, _ v: Any?) {
        self.n = n

        if let v = v as? AnyCodable {
            self.v = v
        } else {
            self.v = AnyCodable(v)
        }
    }

    public func encode(to encoder: Encoder) throws {
        // A parameter without a value keeps its name and travels as null, which the synthesized encoding would leave out.

        var container = encoder.container(keyedBy: CodingKeys.self)

        try container.encode(n, forKey: .n)

        if let v {
            try container.encode(v, forKey: .v)
        } else {
            try container.encodeNil(forKey: .v)
        }
    }
}

File: python/lib/src/xplpc/message/param.py
from typing import Any


class Param:
    def __init__(self, name: str, value: Any):
        self.name = name
        self.value = value

    def to_json(self) -> dict[str, Any]:
        value = self.value.to_json() if hasattr(self.value, "to_json") else self.value
        return {
            "n": self.name,
            "v": value,
        }
